File: core/src/main/java/com/roboxue/niffler/javaDSL/Formula.java
package com.roboxue.niffler.javaDSL;

import com.google.common.collect.ImmutableList;
import com.roboxue.niffler.Token;

/**
 * @author robert.xue
 * @since 2019-04-01
 */
public interface Formula<T> {
    ImmutableList<Token<?>> getDependsOn();
    Token<T> getOutlet();
}

File: core/src/main/scala/com/roboxue/niffler/scalaDSL/Formula.scala
package com.roboxue.niffler.scalaDSL
import com.roboxue.niffler.{Token, TokenMeta}

/**
  * @author robert.xue
  * @since 7/15/18
  */
trait Formula[T] {
  val dependsOn: Seq[Token[_]]
  val outlet: Token[T]
}
